lab_1/scripts/update-pipeline-definition.sh

```diff
@@ -0,0 +1,86 @@
+if ! command -v jq  &> /dev/null
+then
+    echo "jq could not be found, but can be easely installed with sudo"
+    exit 0
+fi
+
+if [[ !($1 =~ pipeline\.json$) ]]
+then
+echo "Error: no path to pipeline file is provided"
+exit 1
+fi
+
+if [[ $(jq '.pipeline.version' $1) == null ]]
+then
+    echo "Error: no .version field"
+    exit 1
+fi
+if [[ $(jq '.pipeline.stages[1].actions[0].configuration.EnvironmentVariables' $1) == null ]]
+then
+    echo "Error: no .EnvironmentVariables field for stage 1"
+    exit 1
+fi
+if [[ $(jq '.pipeline.stages[3].actions[0].configuration.EnvironmentVariables' $1) == null ]]
+then
+    echo "Error: no .EnvironmentVariables field for stage 2"
+    exit 1
+fi
+if [[ $(jq '.metadata' $1) == null ]]
+then
+    echo "Error: no .metadata field"
+    exit 1
+fi
+if [[ $(jq '.pipeline.stages[0].actions[0].configuration.Branch' $1) == null ]]
```

Comment:
you stick to particular stage and particular action index.

There is no action name filtering (**Source** action should be changed only):

"The **Branch** property in the **Source** action’s configuration is set to a value from the script’s parameter/flag --branch. "